x/wasm/keeper/relay.go

```diff
@@ -145,11 +145,7 @@ func (k Keeper) OnRecvPacket(
 	// emit all events from this contract itself
 	events := types.ParseEvents(res.Attributes, contractAddr)
 	ctx.EventManager().EmitEvents(events)
-
-	if err := k.dispatchAll(ctx, contractAddr, contractInfo.IBCPortID, res.Submessages, res.Messages); err != nil {
-		return nil, err
-	}
-	return res.Acknowledgement, nil
+	return k.wasmVMResponseHandler.Handle(ctx, contractAddr, contractInfo.IBCPortID, res.Submessages, res.Messages, res.Acknowledgement)
```

Comment:
:100: amazing how you made this even simpler in this case (with the proper acknowledgement overwrite)